include/nbl/builtin/hlsl/central_limit_blur/box_blur.hlsl

```diff
@@ -0,0 +1,297 @@
+#ifndef _NBL_BUILTIN_HLSL_CENTRAL_LIMIT_BLUR_BOX_BLUR_INCLUDED_
+#define _NBL_BUILTIN_HLSL_CENTRAL_LIMIT_BLUR_BOX_BLUR_INCLUDED_
+
+#include <nbl/builtin/hlsl/central_limit_blur/common.hlsl>
+
+#include <nbl/builtin/hlsl/glsl_compat/core.hlsl>
+#include <nbl/builtin/hlsl/workgroup/arithmetic.hlsl>
+#include <nbl/builtin/hlsl/workgroup/scratch_size.hlsl>
+
+
+// https://github.com/microsoft/DirectXShaderCompiler/issues/6144
+static nbl::hlsl::uint32_t3 /*nbl::hlsl::glsl::*/gl_WorkGroupSize() { return uint32_t3( WORKGROUP_SIZE, 1, 1 ); }
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace central_limit_blur
+{
+
+static const uint32_t arithmeticSz = nbl::hlsl::workgroup::scratch_size_arithmetic<WORKGROUP_SIZE>::value;
+static const uint32_t broadcastSz = nbl::hlsl::workgroup::scratch_size_broadcast;
+static const uint32_t scratchSz = arithmeticSz + broadcastSz;
+
+groupshared uint32_t scratch[ scratchSz ];
+
+template<typename T, uint16_t offset>
+struct ScratchProxy
+{
+	T get( const uint32_t ix )
+	{
+		return nbl::hlsl::bit_cast< T, uint32_t >( scratch[ ix + offset ] );
+	}
+	void set( const uint32_t ix, NBL_CONST_REF_ARG( T ) value )
+	{
+		scratch[ ix + offset ] = nbl::hlsl::bit_cast< uint32_t, T >( value );
+	}
+
+	void workgroupExecutionAndMemoryBarrier()
+	{
+		nbl::hlsl::glsl::barrier();
+	}
+};
+
+ScratchProxy<float32_t, 0> prefixSumsAccessor;
+ScratchProxy<float32_t, arithmeticSz> broadcastAccessor;
+
+struct prefix_sum_t
+{
+	using bin_op_t = nbl::hlsl::plus<float32_t>;
+	using type_t = bin_op_t::type_t;
+
+	type_t operator()( NBL_CONST_REF_ARG( type_t ) value )
+	{
+		type_t retval = nbl::hlsl::workgroup::inclusive_scan<bin_op_t, arithmeticSz>::template __call<ScratchProxy<type_t, 0> >( value, prefixSumsAccessor );
+		// we barrier before because we alias the accessors for Binop
+		prefixSumsAccessor.workgroupExecutionAndMemoryBarrier();
+		return retval;
+	}
+};
+
+prefix_sum_t workgroupPrefixSum;
```

Comment:
don't leave globals like this laying around in headers, also the acessor needs to be configurable, so there's literally nothing you do here, except choose your bin_op and type_t 

But then this wrapper needs to move into a general struct `Blur` that encompasses everything